services/management/dbaas/components_service.go

```diff
@@ -458,3 +462,35 @@ func (c componentsService) InstallOperator(ctx context.Context, req *dbaasv1beta
 
 	return &dbaasv1beta1.InstallOperatorResponse{Status: dbaasv1beta1.OperatorsStatus_OPERATORS_STATUS_OK}, nil
 }
+
+// LatestRecommended returns the latest recommended version from a list of product versions or, if there are no
+// recommended versions, returns the latest.
+// Since dbaasv1beta1.Component doesn't have the version number as part of the structure, we also return the
+// version string. This string is used in other places to identify the version.
+func LatestRecommended(m map[string]*dbaasv1beta1.Component) (string, *dbaasv1beta1.Component, error) {
```

Comment:
we don't need to go through all versions and compare them, we can just select the one with `true` value in `Default` field.